CI: Travis flake on arm64: StoreConsulSuite.TestStoreLocalKeyProtection: "timeout while waiting for expected value"

Seen this travis CI flake in unit tests on master during v1.8 dev cycle:

https://travis-ci.com/github/cilium/cilium/jobs/299021385#L1601

In this case it only failed on arm64 build, but passed on amd64 build.

```
=== RUN   Test

{"level":"warn","ts":"2020-03-17T17:21:24.079Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-56e7cff1-a5d3-47bc-a34a-5aec1a7c5203/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:21:24.079Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-56e7cff1-a5d3-47bc-a34a-5aec1a7c5203/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:21:24.138Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-93f385f7-95b1-44fc-b9bf-e12c7b889438/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:21:24.138Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-93f385f7-95b1-44fc-b9bf-e12c7b889438/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:21:24.195Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-1446e1c1-0230-4d19-a093-104fe90b75e8/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:21:24.196Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-1446e1c1-0230-4d19-a093-104fe90b75e8/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:22:24.251Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-4751680e-f781-4437-90e1-575a67b98de3/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:22:24.251Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-4751680e-f781-4437-90e1-575a67b98de3/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:22:24.296Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-3bcf90a0-34af-4494-80b2-f08e766925c7/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-03-17T17:22:24.297Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-3bcf90a0-34af-4494-80b2-f08e766925c7/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

----------------------------------------------------------------------

FAIL: <autogenerated>:1: StoreConsulSuite.TestStoreLocalKeyProtection

store_test.go:270:

    c.Assert(expect(func() bool {

        v, err := kvstore.Client().Get(context.TODO(), store.keyPath(&localKey1))

        return err == nil && v == nil

    }), IsNil)

... value *errors.errorString = &errors.errorString{s:"timeout while waiting for expected value"} ("timeout while waiting for expected value")

OOPS: 9 passed, 1 FAILED

--- FAIL: Test (338.46s)

FAIL

coverage: 8.7% of statements in ./...

FAIL	github.com/cilium/cilium/pkg/kvstore/store	338.792s

FAIL

Makefile:146: recipe for target 'unit-tests' failed

make: *** [unit-tests] Error 1

The command "./.travis/build.sh" exited with 2.
```

--- a/pkg/kvstore/store/store_test.go
+++ b/pkg/kvstore/store/store_test.go
@@ -267,9 +267,10 @@ func (s *StoreSuite) TestStoreLocalKeyProtection(c *C) {
 	c.Assert(expect(func() bool { return localKey1.updated() >= 1 }), IsNil)
 	// delete all keys
 	kvstore.Client().DeletePrefix(context.TODO(), store.conf.Prefix)
+	time.Sleep(10 * time.Millisecond)
 	c.Assert(expect(func() bool {
 		v, err := kvstore.Client().Get(context.TODO(), store.keyPath(&localKey1))
-		return err == nil && v == nil
+		return err == nil && v != nil
 	}), IsNil)
 }
 

